### ISSUE TITLE
Docs: Describe how to install canary versions

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6046]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

We mention in https://github.com/typescript-eslint/typescript-eslint/tree/a4f85b8cfe38ba8ea2a2ac4a56d9b11a81a8a15a#versioning that you can download and install the latest nightly release in the `canary` branch. But we never actually say how to do that.

Trying out locally, I find myself having to use a `--force` flag when installing with npm:

```shell
npm i @typescript-eslint/eslint-plugin@canary @typescript-eslint/parser@canary --save-dev --force
```

Yarn doesn't seem to need that:

```shell
yarn add @typescript-eslint/eslint-plugin@canary @typescript-eslint/parser@canary --save-dev
```

We should document this on the site. Especially for situations like #5688 where many people want to use the canary version, it should be clear for people how to.

### Affected URL(s)

Right now, https://github.com/typescript-eslint/typescript-eslint#versioning. After #5984, perhaps an independent page?
